app/lib/comakery/api_signature.rb

```
@@ -164,7 +166,7 @@ def verify_http_method
       end
 
       def verify_nonce
-        raise Comakery::APISignatureError, 'Invalid nonce' unless @is_nonce_unique.call(@request.fetch('body', {}).fetch('nonce', ''))
+        raise Comakery::APISignatureError, 'Invalid nonce' unless @is_nonce_unique.call(normalize_nonce(@request.fetch('body', {}).fetch('nonce', '')))
```

We should probably return an error if the nonce size is > MAX_NONCE_SIZE. Otherwise the behavior could be odd could be odd for API users. 

For example if I used a long prefix for the nonce it would truncate the random part of the nonce. Then I would get intermittent nonce failures even though I was generating sufficiently random nonces on the client side.